File: src/main/java/org/pl/lexer/tokenizer/PlusTokenizer.java
package org.pl.lexer.tokenizer;

import org.pl.lexer.ILexer;
import org.pl.lexer.token.arithmetic.PlusToken;

public class PlusTokenizer implements ITokenizer {
    @Override
    public boolean matches(Character c) {
        return c == '+';
    }

    @Override
    public TokenizerResult tokenize(ILexer lexer) {
        return new TokenizerResult(new PlusToken());
    }
}

File: src/main/java/org/pl/lexer/tokenizer/PlusTokenizer.kt
package org.pl.lexer.tokenizer

import org.pl.lexer.ILexer
import org.pl.lexer.token.arithmetic.PlusToken

/**
 * Tokenizes the plus symbol '+'.
 */
class PlusTokenizer : ITokenizer {
    override fun matches(c: Char) = c == '+'
    override fun tokenize(lexer: ILexer) = PlusToken()
}